studio/src/ui/popoverMenus/Popover.js

```diff
@@ -58,14 +60,18 @@ export class Popover {
 		}
 	}
 
+	/**
+	 * @type {((...args: any[]) => void) | null}
+	 */
+	initialize = null;
```

Comment:
Maybe it would be a better move (from an OOP perspective) to remove the `initialize` function, and instead integrate the init args into the subclass constructor.

Reply:
I feel like I've tried this before and couldn't easily make it work. But I don't remember what the reason was. Though the situation is probably different now. If you can make this work I'd say go for it!
That could eliminate the need for `createContextMenu` entirely. Or at least very much simplify it, I'm not sure how easy it will be to import the `ContextMenu` constructor everywhere in regards to circular imports.

Reply:
> Maybe it would be a better move (from an OOP perspective) to remove the `initialize` function, and instead integrate the init args into the subclass constructor.

Hm ok. I faffed around with this for a bit, and I definitely get what you mean when you say you "couldn't easily make it work." It's a tough cookie to crack!

I don't think the abstract function approach you mentioned would work because needing to add params in an overridden function violates some core OOP principles (IIRC).

Would it be alright to stick with the `initialize = null` for now? That seemed to work (even if it is a bit sloppy). I think I need to kinda rack my brain a bit more on this issue before I fully tackle it. 

If you are interested you can [take a look at the approach I was aiming for](https://github.com/maxmmyron/Renda/pull/1). It's a bit sloppy since I ended up scrapping most of it 😥

Reply:
Sure I can have a go at it. I think I've done this before so I feel like it should be possible. But I'll probably end up coming to the same conclusion as you.
Let me know when you're done, then I'll review the changes and I'll try it out after that. Otherwise we'll surely run into conflicts.

As for `initialize = null`, I just realised you can just remove `initizale` from the base class, since it is never called from the base class anyway. That way you don't have to override anything either.

Reply:
I finally managed to make some time to look at this. It's over at https://github.com/rendajs/Renda/pull/624 in case you want to check it out.